Frameworks/CoreFoundation/Locale.subproj/CFCalendar.c

```diff
@@ -746,6 +746,12 @@ static CFRange __CFCalendarGetRangeOfUnit2(CFCalendarRef calendar, CFCalendarUni
     // WINOBJC: Add cases for weekOf units. ICU will handle them below.
     case kCFCalendarUnitWeekOfYear:
     case kCFCalendarUnitWeekOfMonth:
+    // WINOBJC: Asking for the range of WeekOfMonth in Year always returns 6.
```

Comment:
It seems to be an issue with ICU. Since ICU is operating on a particular date, and that date happens to not have 6 weeks in it, ICU will not update weekOfMonth to 6. It's essentially an out of bounds date component. Setting the smaller unit to week would instead return how many weeks are in the year. Even if we changed the unit to add to week, it would still not update to 6 weeks if the month does not have that because again, it's an out of bounds value for the unit of WeekOfMonth. To begin with, it doesn't make much sense to ask how many weeks are in a month in a year. I'm merely matching what the reference platform does in this case. I'd consider this a very low priority fringe case that doesn't provide much of a basis for changing core foundation code.

---
In reply to: [74354002](https://github.com/Microsoft/WinObjC/pull/844#discussion_r74354002) [](ancestors = 74354002,74342094,74336418)